libraries/app/include/graphene/app/api.hpp

```diff
@@ -298,8 +298,28 @@ namespace graphene { namespace app {
           */
          std::vector<net::potential_peer_record> get_potential_peers() const;
 
+         /**
+          * @brief Return list of pending transactions.
+          */
+         std::vector<signed_transaction> list_pending_transactions() const;
+
+         /**
+          * @brief Subscribes caller for notifications about pending transactions.
+          * @param callback a functional object which will be called when new transaction is created. 
+          */
+         void subscribe_to_pending_transactions(std::function<void(const variant&)> callback);
+
+         /**
+          * @brief Unsubscribes caller from notifications about pending transactions.
+          */
+         void unsubscribe_from_pending_transactions();
+
       private:
          application& _app;
+         std::vector< signed_transaction > _pending_transactions;
```

Comment:
I'd rather use a `map< transaction_id_type, signed_transaction >` here for better performance.

Reply:
Good hint .. thank you!